src/rpm/package.rs

```diff
@@ -175,7 +222,25 @@ impl RPMPackage {
 
         verifier.verify(header_and_content_cursor, signature_header_and_content)?;
 
-        Ok(())
+        Ok(SignatureVerificationOutcome::Pass)
+    }
+
+    pub fn verify_digest(&self) -> Result<DigestVerificationOutcome, RPMError> {
+        let mut header = Vec::<u8>::with_capacity(1024);
+        // make sure to not hash any previous signatures in the header
+        self.metadata.header.write(&mut header)?;
+
+        let recreated_from_content = Self::create_digests_from_readers(
+            &mut header.as_slice(),
+            SeqCursor::new(&[header.as_slice(), self.content.as_slice()]),
+        )?;
+
+        let package_contained = self.metadata.get_digests()?;
+        if recreated_from_content == package_contained {
+            Ok(DigestVerificationOutcome::Match)
+        } else {
+            Ok(DigestVerificationOutcome::Mismatch)
+        }
```

Comment:
A test for this would be good.

Like above though I'm confused about the error strategy.  If I were to write some fairly intuitive code such as

```Rust
pkg.verify_digest()?;
```

it effectively won't actually be checking the digests, and it would be a silent footgun unless the return value was marked as `#[must_use]` and the user checked the value themselves.

But I'm not sure I understand why this is better than `Result<(), RpmError>` with the addition of new error variants for digest mismatch, as we already have one for signature mismatch?

Reply:
Rationale was, that the RPM format is fine, the signature just failed to verify. And the verfication might simply caused by a missing key.

But I think for the common usage pattern you're correct that it poses a footgun.

Reply:
`VerificationError` could have subtypes potentially

Reply:
Actually there is `KeyNotFound` error already https://github.com/rpm-rs/rpm/pull/122/files#diff-6ff19a1ff611f3835b5a025663cf4dec901ed914626f78847414836439bd1a79L77